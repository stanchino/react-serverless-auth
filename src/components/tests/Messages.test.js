import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Messages } from "..";

const mockStore = configureStore();

const subject = store => (mount(<Provider store={store}><Messages /></Provider>));

const matchFlash = (state, text) => {
    describe("when the flash is set", () => {
        const store = mockStore(state);
        it("shows the message", () => {
            expect(subject(store).text()).toMatch(text);
        })
    });
};

describe("Flash Component", () => {
    it("matches the snapshot", () => {
        const store = mockStore({ auth: { flash: { error: null, notice: null } } });
        expect(renderer.create(<Provider store={store}><Messages /></Provider>).toJSON()).toMatchSnapshot();
    });
    matchFlash({ auth: { flash: { error: 'error' } } }, "error");
    matchFlash({ auth: { flash: { notice: 'notice' } } }, "notice");
});