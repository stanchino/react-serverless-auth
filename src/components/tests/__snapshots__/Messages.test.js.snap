// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flash Component matches the snapshot 1`] = `
<div
  className="messages"
>
  <div
    className="flash"
  />
</div>
`;
